src/test/java/picard/sam/BamErrorMetric/BaseErrorAggregationTest.java

```diff
@@ -0,0 +1,192 @@
+        Assert.assertEquals(metric1.TOTAL_BASES, length);
+        Assert.assertEquals(metric1.NUM_BASES_WITH_OVERLAPPING_READS, length);
+        Assert.assertEquals(metric1.NUM_DISAGREES_WITH_REFERENCE_ONLY, 2L);
+        Assert.assertEquals(metric1.NUM_DISAGREES_WITH_REF_AND_MATE, 1L);
+        Assert.assertEquals(metric1.NUM_THREE_WAYS_DISAGREEMENT, 1L);
+
+        final OverlappingErrorMetric metric2 = (OverlappingErrorMetric) metrics[1];
+        metric2.calculateDerivedFields();
+        Assert.assertEquals(metric2.COVARIATE, ReadBaseStratification.ReadDirection.NEGATIVE.toString());
+        Assert.assertEquals(metric2.TOTAL_BASES, length);
+        Assert.assertEquals(metric2.NUM_BASES_WITH_OVERLAPPING_READS, length);
+        Assert.assertEquals(metric2.NUM_DISAGREES_WITH_REFERENCE_ONLY, 2L);
+        Assert.assertEquals(metric2.NUM_DISAGREES_WITH_REF_AND_MATE, 1L);
+        Assert.assertEquals(metric2.NUM_THREE_WAYS_DISAGREEMENT, 1L);
+    }
+
+    @Test
+    public void testBaseErrorAggregation3() {
+        final SAMSequenceRecord samSequenceRecord = new SAMSequenceRecord("chr1", 200);
+        final SAMFileHeader samFileHeader = new SAMFileHeader();
+        samFileHeader.addSequence(samSequenceRecord);
+
+        final SAMRecordSetBuilder builder = new SAMRecordSetBuilder();
+
+        final List<SAMRecord> samRecords = builder.addPair("Read1234", 0, 1, 1,
+                false, false, "16M", "16M", true, false, 20);
+        final SAMRecord samRecord1 = samRecords.get(0);
+        final SAMRecord samRecord2 = samRecords.get(1);
+
+        samRecord1.setReadBases("CgTGtGGAcAAAgAAA".getBytes());
+        samRecord2.setReadBases("CcTGGtGAcAAAgAAA".getBytes());
+        final byte[] refBases = "CATGGGGAAAAAAAAA".getBytes();
+
+        BaseErrorAggregation<?> baseErrorAggregation = new BaseErrorAggregation<>(SimpleErrorCalculator::new, ReadBaseStratification.referenceBaseStratifier);
+        final int length = refBases.length;
+
+        for (int i = 0; i < length; i++) {
+
+            SamLocusIterator.LocusInfo locusInfo = new SamLocusIterator.LocusInfo(samSequenceRecord, i + 1);
+            SamLocusIterator.RecordAndOffset recordAndOffset1 = new SamLocusIterator.RecordAndOffset(samRecord1, i);
+            SamLocusIterator.RecordAndOffset recordAndOffset2 = new SamLocusIterator.RecordAndOffset(samRecord2, i);
+            locusInfo.add(recordAndOffset1);
+            locusInfo.add(recordAndOffset2);
+
+            final SAMLocusAndReference locusAndReference = new SAMLocusAndReference(locusInfo, refBases[i]);
+
+            baseErrorAggregation.addBase(recordAndOffset1, locusAndReference);
+            baseErrorAggregation.addBase(recordAndOffset2, locusAndReference);
+        }
+
+        final ErrorMetric[] metrics = baseErrorAggregation.getMetrics();
+        final BaseErrorMetric metricA = Arrays.stream(metrics).map(a -> (BaseErrorMetric) a)
+                .filter(a -> a.COVARIATE.equals("A")).findFirst().orElse(null);
```

Comment:
The code calls `orElse(null)` and on the next line makes a method call, generating a NPE. This is OK for a test but IMO using `.get()` would a simpler way to write it.